### PR TITLE
Syringe gun!

### PR DESCRIPTION
**Pull request by ScarKy0**
## About the PR

Feature for #24773 
The idea is that it will slowly inject the syringes over time, giving time to react and pull them out.
Balance very much not done, but ideas very much appreciated.

## Why / Balance

It cool.

## Technical details

Adds a new SolutionInjectWhileEmbedded component that does exactly what it sounds like. (Injects solution at rate and amount specified in the component)

## Media

https://github.com/user-attachments/assets/800ead03-6b94-487f-809b-48c71389130f

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added the syringe gun and it's respective ammo. Currently Admeme only.

### CODE REVIEW

**Comment by Dutch-VanDerLinde** [comment]
Give the syringe this component
  - type: ThrowingAngle
    angle: 0
    
    so that the syringe appears straight when its fired

**Comment by ScarKy0** [comment]
> Give the syringe this component
type: ThrowingAngle
angle: 0
so that the syringe appears straight when its fired

Guess what, 45 makes it straight for throwing but when fired they will ALWAYS take this angle no matter what, will have to be fixed seperately

**Comment by Dutch-VanDerLinde** [comment]
it shouldn't? arrow has this behaviour so maybe check the arrow's code

**Comment by ScarKy0** [comment]
> it shouldn't? arrow has this behaviour so maybe check the arrow's code

Because arrow has a sprite that points straight. Syringe sprite is at a 45 degree angle

**Comment by Dutch-VanDerLinde** [comment]
simply make the sprite not at a 45 degree angle

**Comment by ScarKy0** [comment]
> simply make the sprite not at a 45 degree angle

Lemmie just redo all the syringe sprites really quick :godo:

**Comment by Boaz1111** [comment]
this should probably just use a different thing

**Comment by Boaz1111** [comment]
like a new syringe ammo

**Comment by ScarKy0** [comment]
> like a new syringe ammo

Thats the idea for now, to add craftable minisyringes or something, but for now I was asked to make it use regular syringes until we start balancing stuff

**Comment by Kadeo64** [comment]
syringe gun is finally real

**Comment by foob1** [comment]
syringe gun? from the hit game barotrauma? like THE syringe gun? from barotrauma?

**Comment by Aisu9** [comment]
isn't the syringe gun supposed to inject after a delay and not instantly like the issue linked state?

**Comment by ScarKy0** [comment]
> I wouldn't be too worried about the potassium-water combo since you either need to connect two shots without missing or get a cryostasis syringe

Yeah I think its fine too, if anything balancing comes after this even works

**Comment by Kadeo64** [comment]
cannot wait to shoot bluespace syringes of weh juice to people

**Comment by ScarKy0** [comment]
>cannot wait to shoot bluespace syringes of weh juice into people

It wont inject fast enough to make much of a differance, unless It will inject a percentage of its contents per second

**Comment by Kadeo64** [comment]
> It wont inject fast enough to make much of a differance, unless It will inject a percentage of its contents per second

taking a syringe out requires a doafter like an embedded arrow, right? just shoot multiple syringes and they can't deal with that :trollface:
add a little bit of noct to like the third syringe and boom free inject

**Comment by ScarKy0** [comment]
> taking a syringe out requires a doafter like an embedded arrow, right?

true :trollface:

**Comment by thebadman4662** [comment]
> > It wont inject fast enough to make much of a differance, unless It will inject a percentage of its contents per second
> 
> taking a syringe out requires a doafter like an embedded arrow, right? just shoot multiple syringes and they can't deal with that :trollface: add a little bit of noct to like the third syringe and boom rest of the syringes inject fully

Fill it with Lico/Taz, infinite shock therapy :trollface:

**Comment by foob1** [comment]
barotrauma reference

**Comment by Kadeo64** [comment]
> barotrauma reference

referenception

**Comment by Aisu9** [comment]
> taking a syringe out requires a doafter like an embedded arrow, right?

following the discussion, removing syringes should be instantaneous

**Comment by Kadeo64** [comment]
> following the discussion, removing syringes should be instantaneous

as to not cause 300 nerfs in the future, this, AND syringes embedded need to be limited. You should only be able to have ~3 syringes embedded in you at once and if another one hits you it throws the first one out. this way you can't just spam a thousand syringes into someone that they can't react to

**Comment by ScarKy0** [comment]
It's done!
Except balance...

**Comment by ScarKy0** [comment]
I swear it works i dunno why the github check gods chose to target me

File: Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs:33]: Empty method.

File: Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs:53]: You should never do curtime + interval that causes rounding problems, just add on the update interval.

Additionally this should be set on MapInit, also these need to be cleaned up or they will be a performance sink for the rest of a round.

File: Resources/Prototypes/Entities/Objects/Specific/chemistry.yml
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Specific/chemistry.yml:445]: This feels really duplicated for no reason.

File: Resources/Prototypes/Entities/Objects/Specific/chemistry.yml
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Specific/chemistry.yml:404]: This does nothing remove it please.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Audio isn't attributed.

File: Resources/Prototypes/Entities/Objects/Specific/chemistry.yml
Comment (ScarKy0) [reply Resources/Prototypes/Entities/Objects/Specific/chemistry.yml:445]: What part is duplicated here? Projectile so they can collide and fixtures so they dont go through walls
I dont think that stuff can be removed

**Comment by Bobylein** [comment]
> > following the discussion, removing syringes should be instantaneous
> 
> as to not cause 300 nerfs in the future, this, AND syringes embedded need to be limited. You should only be able to have ~3 syringes embedded in you at once and if another one hits you it throws the first one out. this way you can't just spam a thousand syringes into someone that they can't react to


it's a singlefire syringe gun, make loading it a doafter of 1-2 seconds and when a lot of people cooperate to turn you into a pincushion I feel they should be rewarded for that.

File: Resources/Prototypes/Entities/Objects/Specific/chemistry.yml
Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Objects/Specific/chemistry.yml]: What does it even have 2 fixtures for.

Comment (ScarKy0) [reply Resources/Prototypes/Entities/Objects/Specific/chemistry.yml]: One for projectile one for the actual item
Remove projectile, it doesnt hit anyone
Remove fix1, it goes through walls
Both are needed to function, so is the projectile component because it's parent is the syringe not an arrow or something

File: Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs
Comment (ScarKy0) [reply Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs]: Can you elaborate on what "Cleaned up" means?
As in do something on component shutdown?
If destroyed then I don't think they can be since there will be a limited amount of them provided anyways, would be ass to only have like 3 and lose them after use. (Unless it ends up being available to all crew and not only traitors)

File: Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs:45]: Just use autopausedfield.

File: Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs:61]: styleguide please.

File: Content.Server/Chemistry/Components/SolutionInjectWhileEmbeddedComponent.cs
Comment (metalgearsloth) [inline Content.Server/Chemistry/Components/SolutionInjectWhileEmbeddedComponent.cs:24]: styleguide

File: Content.Shared/Projectiles/SharedProjectileSystem.cs
Comment (metalgearsloth) [inline Content.Shared/Projectiles/SharedProjectileSystem.cs:133]: Component is never dirtied so client never gets this field, same above.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Chemistry/Components/SolutionInjectWhileEmbeddedComponent.cs
Comment (metalgearsloth) [inline Content.Server/Chemistry/Components/SolutionInjectWhileEmbeddedComponent.cs:12]: fix indent.

**Comment by ScarKy0** [comment]
> How is this obtained?

It's not, was waiting for maintainer decision but was never given one to implement
So kinda admeme only for now

**Pull request merged**
